feat(KNO-5249): Add Docs for New Relic (#404)
knocklabs/docs: content/integrations/extensions/new-relic.mdx

---
title: Connecting Knock to your New Relic account
description: Learn more about how to connect Knock with your New Relic account.
layout: integrations
tags: ["new relic", "extensions"]
section: Integrations > Extensions
---

You can use the Knock + New Relic integration to stream workflow, channel, and message metrics from your Knock account to your New Relic account. With this you can:

- Set up custom New Relic monitors and dashboards to track your Knock workflows & channels
- Get up-to-the-minute data on workflows triggered and messages delivered
- Monitor events ingested and actions triggered from event platforms like [Segment](/integrations/sources/segment) and [Rudderstack](/integrations/sources/rudderstack)

## What this integration does

> This integration will send a stream of metrics as they happen from your Knock account to your New Relic account. Metrics are prefixed `knock.*` and include success and failure codes. Metrics are tagged (where applicable) by:

Reviewer: ```suggestion
This integration will stream metrics from your Knock account to your New Relic account. Metrics are prefixed `knock.*` and include success and failure codes. Metrics are tagged (where applicable) by:
```

- Environment
- Workflow key
- Workflow exec mode
- Channel or workflow step type
- Integration source type
- Error reason

Please refer to your New Relic pricing agreement for information on how custom metrics sent to New Relic are priced for your account.

> At this time there is no way to selectively enable specific metrics, however metrics will only be emitted to New Relic for features that you are actively using in Knock.

Reviewer: ```suggestion
At this time there is no way to selectively enable specific metrics, but metrics will only be emitted to New Relic for features that you are actively using in Knock.
```

## Installing the integration

1. Visit the integrations page in your Knock dashboard, and click the "Extensions" tab.
2. Click "Configure New Relic".
3. Enter a New Relic API Key from New Relic's [API Keys](https://one.newrelic.com/launcher/api-keys-ui.api-keys-launcher) page (we recommend creating a dedicated key just for Knock).
4. Pick the correct site for your New Relic data hosting (visit [New Relic's docs](https://docs.newrelic.com/docs/accounts/accounts-billing/account-setup/choose-your-data-center/) for more information).
5. Click "Connect".

<Callout
  emoji="🔦"
  text={
    <>

>       When creting an API key, make sure "Key Type" is marked as{" "}

Reviewer: ```suggestion
      When creating a New Relic API key, make sure "Key Type" is marked as{" "}
```
